packages/redis/src/Adaptor.js

```diff
@@ -0,0 +1,232 @@
+import { expandReferences } from '@openfn/language-common/util';
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+
+import { createClient } from 'redis';
+import * as util from './Utils';
+
+let client = null;
+
+export const setMockClient = fakeClient => (client = fakeClient);
+
+const connect = async state => {
+  const {
+    username,
+    password,
+    host,
+    port = 16552,
+    database,
+  } = state.configuration;
+
+  client = await createClient({
+    username,
+    password,
+    database,
+    socket: { host, port },
+  })
+    .on('error', err => {
+      console.error('Redis Client Error', err);
+      throw err;
+    })
+    .connect();
+
+  if (client.isReady) console.log('Client is connected');
+
+  return state;
+};
+
+const disconnect = async state => {
+  await client.disconnect();
+  if (!client.isReady) console.log('Client is disconnected');
+  return state;
+};
+
+/**
+ * Options provided to the scan function
+ * @typedef {Object} ScanOptions
+ * @public
+ * @property {integer} [cursor=0] - A numeric value used to continue the iteration from where it left off. Initially, you start with 0.
+ * @property {string} [type='hash'] - Limits the keys returned to those of a specified type (e.g., string, list, set, hash, etc.).
+ * @property {integer} count - A hint to the server about how many elements to return in the call (default is 10).
+ */
+/**
+ * State object
+ * @typedef {Object} RedisState
+ * @property data - the parsed response body
+ * @property references - an array of all previous data objects used in the Job
+ **/
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute`, creates and disconnect client
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  return state => commonExecute(connect, ...operations, disconnect)(state);
+}
+
+/**
+ * Get the string value of a key
+ * If the key does not exist the special value nil is returned.
```

Comment:
Oh that's interesting, what  is the special value `nil`? Is it a string?

This makes me a bit nervous. Can we have a unit test to reproduce it?

Reply:
I have update the jsdoc to say `null`